xlb/operator/boundary_masker/aabb_fill.py

```diff
@@ -139,45 +100,46 @@ def kernel_with_distance(
             index = wp.vec3i(i, j, k)
 
             # position of the point
-            pos_bc_cell = self.index_to_position(index)
+            cell_center_pos = self.helper_masker.index_to_position(bc_mask, index)
 
             if solid_mask[i, j, k] == wp.uint8(255) or bc_mask[0, index[0], index[1], index[2]] == wp.uint8(255):
                 # Make solid voxel
                 bc_mask[0, index[0], index[1], index[2]] = wp.uint8(255)
             else:
                 # Find the boundary voxels and their missing directions
-                for l in range(1, _q):
-                    _dir = wp.vec3f(wp.float32(_c[0, l]), wp.float32(_c[1, l]), wp.float32(_c[2, l]))
+                for direction_idx in range(1, _q):
+                    direction_vec = wp.vec3f(wp.float32(_c[0, direction_idx]), wp.float32(_c[1, direction_idx]), wp.float32(_c[2, direction_idx]))
```

Comment:
It could be nice to have a formal way to index direction that are not the center.